### ISSUE TITLE
BUG: generating invalid ocaml syntax.

### ISSUE THREAD

**Issue by nojb** [ocaml-ppx/ocamlformat#2536]
`ocamlformat.0.26.1` chokes on:
```
let _ =
  if x && try () with _ -> () then ()
```
when `.ocamlformat` contains
```
ocp-indent-compat
```

**Comment by Julow** [comment]
Thanks for the report. The bug doesn't occur on the main branch, could you check ? This seems to be fixed since #2445.

**Comment by nojb** [comment]
> Thanks for the report. The bug doesn't occur on the main branch, could you check ? This seems to be fixed since #2445.

I tried master but now I got another BUG :)
```
let _ =
  try
    ()
  with _ ->
    (match () with _ -> ())
```
with
```
indicate-multiline-delimiters=closing-on-separate-line
```

**Comment by nojb** [comment]
And another:

```
let _ =
  try () with _ ->
  try () with _ -> () in
()
```
also with
```
indicate-multiline-delimiters=closing-on-separate-line
```

**Comment by Julow** [comment]
Thanks! This is hopefully a recent bug. Should be fixed in https://github.com/ocaml-ppx/ocamlformat/pull/2537

**Comment by nojb** [comment]
> Thanks! This is hopefully a recent bug. Should be fixed in #2537

Thanks, this fixes it indeed, and the original bug reported is fixed in `master` already. Closing. Thanks!

**Comment by Julow** [comment]
Thanks for checking.
